src/gluonts/mx/distribution/mixture.py

```diff
@@ -60,6 +61,32 @@ def __init__(
         # self.all_same = len(set(c.__class__.__name__ for c in components)) == 1
         self.mixture_probs = mixture_probs
         self.components = components
+        if not isinstance(mixture_probs, mx.sym.Symbol):
+
+            # assert that all components have the same batch shape
+            assert np.all(
+                [d.batch_shape == self.batch_shape for d in components[1:]]
+            ), "All component distributions must have the same batch_shape."
+
+            # assert that mixture_probs has the right shape
+            assertion_message = f"""mixture_probs have shape {mixture_probs.shape}, but expected shape: (..., k), 
+                                    where k is len(components)={len(components)}. 
+                                    All axes except the last one should either coincide with the ones from the 
+                                    component distributions, 
+                                    or be 1 (in which case, the mixing coefficient is shared across
+                                    the axis)."""
+
+            expected_shape = self.batch_shape + (len(components),)
+            assert len(expected_shape) == len(mixture_probs.shape), (
```

Comment:
How come the test cases added in #1042 are not failing here?

Reply:
They are not really probing the edge case:

```python
p=mx.nd.array([0.2]),
mixture_probs = mx.nd.stack(p, 1.0 - p, axis=-1)
mixture_probs.shape
(1,2)
```
=> The shape of the added test cases is correct.